liana-gui/src/app/settings/mod.rs

```diff
@@ -111,6 +114,184 @@ where
+#[derive(Debug, Clone, Deserialize, Serialize)]
+pub struct CubeSettings {
+    pub id: String,
+    pub name: String,
+    pub network: Network,
+    pub created_at: i64,
+    /// The Vault wallet for this Cube (optional - may not be set up yet)
+    pub vault_wallet_id: Option<WalletId>,
+    /// Optional security PIN (stored as Argon2id hash with salt in PHC format)
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub security_pin_hash: Option<String>,
+}
+
+impl CubeSettings {
+    pub fn new(name: String, network: Network) -> Self {
+        Self {
+            id: uuid::Uuid::new_v4().to_string(),
+            name,
+            network,
+            created_at: chrono::Utc::now().timestamp(),
+            vault_wallet_id: None,
+            security_pin_hash: None,
+        }
+    }
+
+    pub fn with_vault(mut self, wallet_id: WalletId) -> Self {
+        self.vault_wallet_id = Some(wallet_id);
+        self
+    }
+
+    pub fn with_pin(mut self, pin: &str) -> Self {
+        self.security_pin_hash = Some(Self::hash_pin(pin));
+        self
+    }
+
+    pub fn has_pin(&self) -> bool {
+        self.security_pin_hash.is_some()
+    }
+
+    pub fn verify_pin(&self, pin: &str) -> bool {
+        if let Some(stored_hash) = &self.security_pin_hash {
+            Self::verify_argon2_pin(pin, stored_hash)
+        } else {
+            // No PIN set, allow access
+            true
+        }
+    }
+
+    /// Hash PIN using Argon2id with a random salt
+    fn hash_pin(pin: &str) -> String {
+        use argon2::{
+            password_hash::{rand_core::OsRng, PasswordHasher, SaltString},
+            Argon2, Params,
+        };
+
+        // Generate a random salt
+        let salt = SaltString::generate(&mut OsRng);
+
+        // Configure Argon2id with reasonable parameters
+        // m_cost: 19456 KiB (19 MiB), t_cost: 2 iterations, p_cost: 1 thread
+        let params = Params::new(19456, 2, 1, None).expect("Valid Argon2 params");
+        let argon2 = Argon2::new(argon2::Algorithm::Argon2id, argon2::Version::V0x13, params);
+
+        // Hash the PIN
+        let password_hash = argon2
+            .hash_password(pin.as_bytes(), &salt)
+            .expect("Failed to hash PIN");
+
+        // Return the PHC string format
+        password_hash.to_string()
+    }
+
+    /// Verify PIN against Argon2id hash
+    fn verify_argon2_pin(pin: &str, hash: &str) -> bool {
+        use argon2::{
+            password_hash::{PasswordHash, PasswordVerifier},
+            Argon2,
+        };
+
+        // Parse the PHC string
+        let parsed_hash = match PasswordHash::new(hash) {
+            Ok(h) => h,
+            Err(_) => return false,
+        };
+
+        // Verify the PIN
+        Argon2::default()
+            .verify_password(pin.as_bytes(), &parsed_hash)
+            .is_ok()
+    }
+}
+
+#[derive(Debug, Clone, Default, Deserialize, Serialize)]
+pub struct Cubes {
+    pub cubes: Vec<CubeSettings>,
```

Comment:
To reduce mental load, we almost definitely don't need a new `Cubes` struct, associated functions can simply use a `Vec<CubeSettings>`